content/posts/rust.md

```diff
@@ -0,0 +1,309 @@
+---
+title: "Rust/PyO3 Support in Pyodide "
+date: 2022-04-15T12:37:52-07:00
+author: "Hood Chatham"
+# author: ["Me", "You"] # multiple authors
+showToc: true
+TocOpen: false
+draft: false
+hidemeta: false
+comments: false
+description: "Getting the pyca/cryptography package working in Pyodide"
+# canonicalURL: "https://canonical.url/to/page"
+disableHLJS: true # to disable highlightjs
+disableShare: false
+hideSummary: false
+searchHidden: true
+ShowReadingTime: true
+ShowBreadCrumbs: true
+ShowPostNavLinks: true
+cover:
+    image: "<image path/url>" # image path/url
+    alt: "<alt text>" # alt text
+    caption: "<text>" # display caption under cover
+    relative: false # when using page bundles set this to true
+    hidden: true # only hide on current single page
+---
+
+<style>
+.post-content.post-content code {
+  font-size: 70%;
+}
+
+code.pkg {
+  background: white;
+  padding: 0px;
+  margin: 0px;
+}
+</style>
+
+We are happy to announce that the next version of Pyodide will ship with the
+newest version of the Cryptography package, including its Rust extension module.
+The Cryptography package was one of the first major packages that included a
+Rust extension module. By including Rust, the Cryptography maintainers have
+forced the Python ecosystem to adopt Rust support. We are no exception. The
+Cryptography package has been one of the most often requested packages in the
+Pyodide issue tracker so it is a priority for us to support it.
+
+See [this talk by the Cryptography
+maintainers](https://www.youtube.com/watch?v=z_Eiy2W0APU) for an interesting
+discussion of their reasons for using Rust and the problems in the ecosystem
+that needed to be fixed before they could use it.
+
+We want to build the Rust extension module for our WebAssembly-based platform.
+Rust has good support for the <code class="pkg">wasm32-unknown-unknown</code>
+target and has popular tools like <code class="pkg">wasm-bindgen</code>.
+However, we need to use the <code class="pkg">wasm32-unknown-emscripten</code>
+compiler target which does not work as well because it requires coordination
+between the Emscripten and Rust projects.
+
+In this blog post I will give a technical description of many of the problems I
+ran into in the process of building Rust extension modules for Pyodide.
+
+
+### Acknowledgements
+
+Thanks to the Cryptography team for making this necessary and for their
+technical advice. Thanks to the <code class="pkg">chrono</code> maintainers and
+the Rust team for reviewing and merging patches to improve Emscripten support.
+Thanks to the PyO3 team for their help and enthusiasm and for adding Emscripten
+tests to their continuous integration. Special thanks to the Emscripten team,
+particularly Sam Clegg, for their technical advice, for merging patches, and for
```

Comment:
Wowa, small world! Long time no see Sam!

Reply:
Nice!  Great to see you name pop up in my feed Alex.  I hope you are good!